proto/info/hardware.proto

```diff
@@ -0,0 +1,171 @@
+  uint64 thresh = 4;  // the value that the Value of the same attribute must reach for the attribute's state to be considered critical.
+  string when_failed = 5; // critical state if no empty
+  uint64 raw_value = 6; // Contains miscellaneous meter readings (where applicable)
+  string type = 7; // Informs the condition of the device e.g., pre-fail, old_age, etc.
+  string attribute_name = 8; // the attribute name
+}
+
+message StorageDiskInfo {
+  reserved 3; // deprecated
+  string disk_name = 1;
+  string wwn = 4;
+  string serial_number = 5;
+  string model = 6; // Intel 123456F
+  string collector_errors = 7; // Reports errors when collecting information. Default and normal value = ""
+  repeated SmartAttr smart_attr = 8; // smartctl output
+}
+
+message USBBusDevnum {
+        // https://elixir.bootlin.com/linux/v6.16.9/source/include/linux/usb.h#L451
+        // https://github.com/gregkh/usbutils/blob/master/lsusb-t.c#L48
+        uint32 bus = 1;
+        // https://elixir.bootlin.com/linux/v6.16.9/source/include/linux/usb.h#L657
+        // https://github.com/gregkh/usbutils/blob/master/lsusb-t.c#L63
+        uint32 devnum = 2;
+}
+
+message USBDevice {
+        optional PCIAddress pci_parent = 1;
+        optional USBBusDevnum usb_parent = 2;
+
+        // these are uint16, protobuf only supports from uint32
+        // https://elixir.bootlin.com/linux/v6.16.9/source/include/linux/usb.h#L984
+        uint32 vendor_id = 3;
+        uint32 product_id = 4;
+
+        USBBusDevnum bus_devnum = 5;
+}
+
+message PCIAddress {
+        // https://github.com/lyonel/lshw/blob/master/src/core/pci.cc#L223
+        uint32 domain = 1;
+        uint32 bus = 2;
+        uint32 device = 3;
+        uint32 function = 4;
+}
+
+message PCIDevice {
+        optional PCIAddress parent_pci_device_address = 1;
+        string driver = 2;
+        PCIAddress address = 3;
+        // https://elixir.bootlin.com/linux/v6.16.9/source/include/linux/pci.h#L347
+        // using uint32 because uint16 is not available in protobuf
+        uint32 vendor_id = 4;
+        uint32 device_id = 5;
+        uint32 revision = 6;
+        uint32 subsystem_id = 7;
+        uint32 class_id = 8; // 3 bytes: (base,sub,prog-if)
+        uint64 subclass_id = 9;
+        // IOMMU group names apparently are strings: https://elixir.bootlin.com/linux/v6.16.9/source/drivers/iommu/iommu.c#L1101
+        string iommu_group = 10;
+}
+
+message SerialPort {
+        optional PCIAddress pci_parent = 1;
+        optional USBBusDevnum usb_parent = 2;
+
+        string ioports = 3;
+        uint64 irq = 4;
+
+        // f.e. /dev/ttyS0
+        string devpath = 5;
+}
+
+message CANDevice {
+        optional PCIAddress pci_parent = 1;
+        optional USBBusDevnum usb_parent = 2;
+
+        string ifname = 3;
+}
+
+message NetworkDevice {
+        optional PCIAddress pci_parent = 1;
+        optional USBBusDevnum usb_parent = 2;
+
+        string ifname = 3;
+        bool wireless = 4;
```

Comment:
Don't we currently separate WiFi and Cellular? (And then there is BT as well which spec.sh doesn't look for AFAIK.)

Reply:
We do have some existing definitions for the config side of the API but these are intermixed with other I/O adapters:
proto/evecommon/devmodelcommon.proto:  PhyIoNetEth = 1;
proto/evecommon/devmodelcommon.proto:  PhyIoNetWLAN = 5;
proto/evecommon/devmodelcommon.proto:  PhyIoNetWWAN = 6;
proto/evecommon/devmodelcommon.proto:  PhyIoNetEthPF = 11;
proto/evecommon/devmodelcommon.proto:  PhyIoNetEthVF = 12;

Note that we haven't defined anything for BT yet.

How  about renaming NETWORK_DEVICE_TYPE_CELLULAR to NETWORK_DEVICE_TYPE_WWAN and leaving out BT for now?